src/Services/Buienradar.php

```diff
@@ -6,20 +6,26 @@
 
 class Buienradar
 {
-    public static function getForecasts(string $latitude, string $longitude): array
+    public static function getForecasts(string $latitude, string $longitude)
     {
         $endpoint = "https://graphdata.buienradar.nl/forecast/json/?lat={$latitude}&lon={$longitude}";
-
-        $response = Http::get($endpoint)->json();
-
-        return collect($response['forecasts'])
-            ->map(function (array $forecast) {
-                return [
-                    'time' => $forecast['datetime'],
-                    'rain' => $forecast['precipitation'],
-                ];
-            })
-            ->take(12)
-            ->toArray();
+        
+        if (Http::get($endpoint)->ok()) {
```

Comment:
Could you switch this around and do an early return if it's not ok instead? Same with the `count($response)`